client/context/Base.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { createContext, useContext, useEffect, useState } from 'react';
+import { formatClubsData, formatEventsData } from "../utils";
+
+interface AppData {
+    clubs: Map<Id, ClubInfo>;
+    events: Map<Id, EventInfo>;
+}
+
+export interface ClubInfo {
+    name: string
+    description: string
+}
+
+export interface EventInfo {
+    name: string
+    description: string
+}
+
+export type Id = number;
+
+export const clubsData = [
```

Comment:
Alternative name

```suggestion
export const clubs = [
```

Reply:
the state variables are already called `clubs` and `events`. we can rename to `CLUBS` and `EVENTS` instead?